File: task2/InputValidation.java
import java.util.Scanner;
import java.util.InputMismatchException;

/**
 * @author Matthias Bartolo
 * @date 02/11/2022
 */
public class InputValidation {
    // Declaring and Initialising Scanner instance
    private static Scanner sc = new Scanner(System.in);

    /**
     * Default Constructor
     */

    public InputValidation() {

    }

    /**
     * Method for getting and validating an integer type input
     */
    public int GetIntInput(String inputText, int minInputRange, int maxInputRange) {
        // Variable which will hold the user input
        int uinput = minInputRange - 1;

        // Looping until the correct user input is inputted
        do {
            // try catch being used to catch exceptions
            try {
                System.out.println("Please Input " + inputText + ":");
                uinput = sc.nextInt();
            } catch (InputMismatchException e) {
                // Setting condition to display error statement
                uinput = minInputRange - 1;
                // Clearing Scanner
                sc.nextLine();
            }

            // Error message is displayed
            if (uinput < minInputRange || uinput > maxInputRange) {
                System.out
                        .println("\nInvalid Input\nInput must be in range of: " + minInputRange + "-" + maxInputRange);
            }
        } while (uinput < minInputRange || uinput > maxInputRange);
        // Returning validated user input
        return uinput;
    }

}

File: task1/InputValidation.cpp
//
// Created by Matthias on 30/12/2022.
//
//Inclusion of headerFile instance
#include "headerFile.h"

//Method, which returns a valid user input, in a specified range
int InputValidation::GetValidInput(int minrange, int maxrange){
    //Declaring variable, which will be used for loop
    bool Loop=true;
    //Declaring variable, which hold the user input
    int userinput;
    //Looping until a valid user input is entered
    while(Loop==true) {
        //Getting user input, and storing in variable
        cout << "\nEnter Choice (Input needs to be in range of "<<minrange<<"-"<<maxrange<<"):"<< endl;
        cin >> userinput;
        //Checking if cin failed, and whether user input is in incorrect range, if so, will enter if statement
        if (cin.fail() || (userinput < minrange || userinput > maxrange)) {
            //Printing out relevant message to the user, and clearing cin
            cout << "\nInvalid Option"<< endl;
            cin.clear();
            //Storing incorrect input, in string buffer, to allow the user to re-enter a choice
            string incorrectInput;
            getline(cin,incorrectInput);
            //Resetting user input and loop condition
            userinput=0;
            Loop = true;
        } else {
            //Setting appropriate condition, to terminate loop
            Loop = false;
        }
    }
    //Returning user input
    return userinput;
}